rust/lance/src/index/vector/ivf.rs

```diff
@@ -1540,4 +1538,68 @@ mod tests {
                 .all(|v| (-2.0 * DIM as f32..0.0).contains(v)));
         }
     }
+
+    #[tokio::test]
+    async fn test_create_ivf_pq_f16() {
```

Comment:
At some point, I think it would be good to make a parametrize tests that checks all combinations of `float data type` x `distance function`, since we have kernels specialized for all those combinations.